var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -63,6 +63,24 @@ class Hiop(CMakePackage, CudaPackage, ROCmPackage):
     variant("ginkgo", default=False, description="Enable/disable ginkgo solver")
     variant("cusolver", default=False, description="Enable/disable cuSovler")
 
+    variant("full_optimizations", default=False, description="Enable/Disable optimizations and release type")
+    depends_on("ginkgo+full_optimizations", when="+ginkgo+full_optimizations")
+    depends_on("ginkgo~full_optimizations", when="+ginkgo~full_optimizations")
+
+    #force optimizations flag to use Release mode
+    conflicts("build_type=Debug", when="+full_optimizations", msg="Use Release mode when with optimizations")
+    conflicts("build_type=MinSizeRel", when="+full_optimizations", msg="Use Release mode when with optimizations")
+    conflicts("build_type=RelWithDebInfo", when="+full_optimizations", msg="Use Release mode when with optimizations")
+    #force no optimizations flag to use RelWithDebInfo mode
+    conflicts("build_type=Release", when="~full_optimizations", msg="Use RelWithDebInfo mode when without optimizations")
+    conflicts("build_type=Debug", when="~full_optimizations", msg="Use RelWithDebInfo mode when without optimizations")
+    conflicts("build_type=MinSizeRel", when="~full_optimizations", msg="Use RelWithDebInfo mode when without optimizations")
+
+    #force dependencies build types for optimizations
+    for pkg in ["raja", "umpire", "magma", "camp"]:
```

Comment:
Ginkgo can be added here instead
```suggestion
    for pkg in ["raja", "umpire", "magma", "camp", "ginkgo"]:
```